src/components/screen-header/HeaderButton.tsx

```diff
@@ -75,6 +77,34 @@ const BaseHeaderButton = ({
   </TouchableOpacity>
 );
 
+export type LargeHeaderButtonProps = Omit<HeaderButtonProps, 'type' | 'text'>;
```

Comment:
I think the pattern with `type` and such is pretty good. And using it would probably work well with reusing the logic for the existing header?

Reply:
@gorandalum Right now `type` is defined as 

```ts
export type ButtonModes =
  | 'back'
  | 'cancel'
  | 'close'
  | 'chat'
  | 'skip'
  | 'status-disruption'
  | 'custom';
```

I don't think most of them makes sense in this context, maybe just `back`, `cancel` and `close`. And even then, I don't know when we should show the back arrow and not, since the new header concept in Figma isn't covering these use cases. Options to extend this are to add `backArrow` as a mode, it's own param `showBackArrow: boolean` or even a `leftIcon`, but I don't know what's best.

This is why I thought it would be ok to skip this complexity in this iteration, until we know what it will be used for.